src/sentry/issues/priority.py

```diff
@@ -93,32 +97,28 @@ def get_priority_for_ongoing_group(group: Group) -> PriorityLevel | None:
     if not features.has("projects:issue-priority", group.project, actor=None):
         return None
 
-    previous_priority_history = (
-        GroupHistory.objects.filter(
-            group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
-        )
-        .order_by("-date_added")
-        .first()
-    )
+    previous_priority_history = GroupHistory.objects.filter(
+        group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
+    ).order_by("-date_added")
```

Comment:
This looks like an unbounded set - do we actually need to load every single GroupHistory record for a group?

Reply:
we're filtering to only the priority-related statuses (there's 3), I don't think there's more we can filter by. We need the second newest entry, if available.